server/middleware/form/process.js

```diff
@@ -97,6 +97,23 @@ const isUnacceptedDataFields = (field) => {
         isFieldComponentOfType(field, 'somu-list');
 };
 
+function checkValidationCondition(validator, caseData) {
+    if (Object.prototype.hasOwnProperty.call(validator, 'condition') && caseData[validator.condition.fieldName] !== validator.condition.value) {
+        return false;
+    }
+    return true;
+}
+
+function getActiveValidators(validationSchemaJson, caseData) {
+    const validationSchema = typeof validationSchemaJson === 'string' ? JSON.parse(validationSchemaJson) : validationSchemaJson;
```

Comment:
Would a try...catch work well here, there is still the edge case that the string would not be valid JSON